src/inference/AdvancedSMC.jl

```diff
@@ -352,6 +352,20 @@ function DynamicPPL.assume(
     return r, 0, vi
 end
 
+using Random
+# avoid Core._apply_iterate
+(m::DynamicPPL.Model)(x::DynamicPPL.AbstractVarInfo)=first(DynamicPPL.evaluate!!(m, Random.GLOBAL_RNG, x))
+(m::DynamicPPL.Model)(x::DynamicPPL.AbstractVarInfo, y::DynamicPPL.AbstractSampler)=first(DynamicPPL.evaluate!!(m, Random.GLOBAL_RNG, x, y))
+(m::DynamicPPL.Model)(x::DynamicPPL.AbstractVarInfo, y::DynamicPPL.AbstractSampler, z::DynamicPPL.AbstractContext)=first(DynamicPPL.evaluate!!(m, Random.GLOBAL_RNG, x, y, z))
+
+# trace down into
+Libtask.trace_into(::DynamicPPL.Model) = true
+Libtask.trace_into(::typeof(DynamicPPL.evaluate_threadsafe!!)) = true
+Libtask.trace_into(::typeof(DynamicPPL.evaluate_threadunsafe!!)) = true
+Libtask.trace_into(::typeof(DynamicPPL._evaluate!!)) = true
+Libtask.trace_into(::typeof(DynamicPPL.tilde_observe)) = true
+Libtask.trace_into(::typeof(DynamicPPL.tilde_observe!!)) = true
```

Comment:
These definitions are quite severe type piracies, in particular the ones for `(m::Model)(...)`. Is the plan to move these definitions to DynamicPPL once tests pass and before this PR is merged?  The definitions of `trace_into` would require that DynamicPPL depends on Libtask though, is this intended?

Reply:
There is a cleaner way to implement the new Libtask-Turing integration. However, for now, we are trying to make things work together. This code will be replaced later.